velox/vector/tests/VectorMakerTest.cpp

```diff
@@ -458,6 +459,58 @@ TEST_F(VectorMakerTest, arrayVectorFromJson) {
   }
 }
 
+TEST_F(VectorMakerTest, nestedArrayVectorFromJson) {
+  auto arrayVector = maker_.nestedArrayVectorFromJson<int32_t>({
+      "[[1, 2], [2, 3, 4], [null, 7]]",
+      "[[1, 3, 7, 9], []]",
+      "[]",
+      "[null]",
+      "[[1, 2, null], [], null]",
```

Comment:
What happens with the last null here ? Shouldn't that expand to std::nullopt as well ?

Reply:
good question... it comes down to how this array of array has been implented under the hood. we create the base vector and then we create the 2d version using offsets (and specifying null rows at this stage) it expands into std::nullopt in the 2d array vector version of it. however, it is not created in the underlying array that is being compared.

at the highest level, the checks from lines 493-509 have checks for that. the underlying 1D array vector contents are compared on line 511.